### PR TITLE
Bootstrap the powertrack-rb gem

### PR DESCRIPTION
**Pull request by lfarcy**
- Setup a gemspec and a Gemfile accordingly
- Use the MIT license and credit @eriwen, gnip-rule's author
- Rule class: immutable instances, validators, unit tests

### CODE REVIEW

File: powertrack-rb.gemspec [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# coding: utf-8
+lib = File.expand_path('../lib', __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'powertrack/version'
+
+Gem::Specification.new do |spec|
+  spec.name          = 'powertrack-rb'
```
Comment (denisj) [inline powertrack-rb.gemspec]: You should call the gem 'powertrack' as long it's a rubygems. But keep the repository named powertrack-rb.

File: powertrack-rb.gemspec [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  spec.version       = PowerTrack::VERSION
+  spec.authors       = ['Laurent Farcy', 'Eric Wendelin']
+  spec.email         = ['laurent.farcy@ecairn.com', 'me@eriwen.com']
+  spec.summary       = %q{Powertrack-rb is a gem used to develop GNIP PowerTrack streaming clients.}
+  spec.description   = spec.summary
+  spec.homepage      = 'https://github.com/ecairn/powertrack-rb'
+  spec.license       = 'MIT license. Copyright eCairn Inc.'
```
Comment (denisj) [inline powertrack-rb.gemspec]: Why using a copyright in he license field?

File: lib/powertrack/rules/string_extension.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require 'powertrack/rules/rule'
+
+# Extend core String class with a rule transformer
+class String
+  def to_pwtk_rule(long=false, tag=nil)
+    PowerTrack::Rule.new(self, long, tag)
+  end
+end
+
```
Comment (denisj) [inline lib/powertrack/rules/string_extension.rb]: Unnecessary line.

File: lib/powertrack/rules/rule.rb [hunk 4]
```diff
@@ -0,0 +1,93 @@
+require 'multi_json'
+
+module PowerTrack
```
Comment (denisj) [inline lib/powertrack/rules/rule.rb:3]: Maybe define a module Rules to follow ruby guidelines?

File: powertrack-rb.gemspec (hunk 2 above)
Comment (lfarcy) [inline powertrack-rb.gemspec]: Removed.

File: powertrack-rb.gemspec (hunk 1 above)
Comment (lfarcy) [inline powertrack-rb.gemspec]: See next commit

File: lib/powertrack/rules/string_extension.rb (hunk 3 above)
Comment (lfarcy) [inline lib/powertrack/rules/string_extension.rb]: Fixed

File: lib/powertrack/rules/rule.rb (hunk 4 above)
Comment (lfarcy) [inline lib/powertrack/rules/rule.rb:3]: I don't feel like there's a guideline. Ruby, unlike Java, does not enforce an automatic module to directory mapping. Here I use the rules directory to include everything related to rules. More to come BTW.

But I still want the Rule class to be available under the PowerTrack module.

**Comment by eriwen** [comment]
:star2:

**Pull request merged**